packages/suite/src/components/suite/modals/confirm/CoinmarketP2pTerms/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Button, Icon, variables, Checkbox, H3 } from '@trezor/components';
+import { Translation, Modal } from '@suite-components';
+import type { Deferred } from '@trezor/utils';
+
+const Text = styled.div`
+    padding-bottom: 20px;
+    text-align: left;
+    & + & {
+        padding-top: 20px;
+        border-top: 1px solid ${props => props.theme.STROKE_GREY};
+    }
+`;
+
+const Header = styled.div`
+    display: flex;
+    width: 100%;
+    padding: 10px 0;
+    align-items: baseline;
+    color: ${props => props.theme.TYPE_DARK_GREY};
+`;
+
+const StyledIcon = styled(Icon)`
+    padding-right: 9px;
+`;
+
+const CheckText = styled.div`
+    font-size: ${variables.FONT_SIZE.BIG};
+`;
+
+const Footer = styled.div`
+    display: flex;
+    padding: 35px 0;
+    flex: 1;
+    border-top: 1px solid ${props => props.theme.STROKE_GREY};
+`;
+
+const FooterContent = styled.div`
+    display: flex;
+    justify-content: space-between;
+    flex: 1;
+    padding: 0 35px;
+`;
+
+export type Props = {
+    provider?: string;
+    decision: Deferred<boolean>;
+    onCancel: () => void;
+};
+
+export const CoinmarketP2pTerms = ({ provider, decision, onCancel }: Props) => {
+    const [isChecked, setIsChecked] = useState<boolean>(false);
```

Comment:
nitpick
extra type declaration, boolean is inferred anyway
```suggestion
    const [isChecked, setIsChecked] = useState(false);
```